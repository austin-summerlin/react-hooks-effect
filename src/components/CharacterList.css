.ul{
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  margin: 10px;
  padding: 10px;
  list-style: none;

}

.li{
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  padding: 10px;
  border: 1px solid black;

}
