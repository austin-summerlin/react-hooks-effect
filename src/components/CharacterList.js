import React from 'react';
import PropTypes from 'prop-types';
import Character from '../components/Character';
import style from '../components/CharacterList.css';

const CharacterList = ({ characters }) => (
  <ul aria-label="character-list" className={style.ul}>
    {characters.map(character => (
      <li key={character.id} className={style.li}>
        <Character
          id={character.id}
          name={character.name}
          image={character.image}
        />
      </li>
    ))}
  </ul>
);

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      image: PropTypes.string,
    })).isRequired
};

export default CharacterList;
