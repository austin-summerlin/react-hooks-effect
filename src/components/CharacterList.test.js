import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from '../components/CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList Component', () => {
  const character = [
    {
      name: 'Test',
      image: 'test.png',
    }
  ];

  afterEach(() => cleanup());
  it('should render the CharacterList component', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList
          characters={character}
        />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
