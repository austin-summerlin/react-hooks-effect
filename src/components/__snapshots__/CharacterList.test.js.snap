// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList Component should render the CharacterList component 1`] = `
<DocumentFragment>
  <ul
    aria-label="character-list"
    class="ul"
  >
    <li
      class="li"
    >
      <a
        class="link-display"
        href="/undefined"
      >
        <img
          alt="Test"
          class="img"
          src="test.png"
        />
        <p
          class="name"
        >
          Test
        </p>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
